### ISSUE TITLE
Deleting a resource from a collection fails

### ISSUE THREAD

**Issue by sblack-usu** [hydroshare/hydroshare#5057]
**Description of the bug**
Resources may not be removed from a collection when relation metadata (that is going to be deleted) is missing.

Steps to reproduce the bug:
1. Copy https://www.hydroshare.org/resource/38feeef698ca484b907b7b3eb84ad05b/
2. Enter edit mode
3. Delete the first resource in the collection, [Example Data II For Surface Water and Ocean Topography Hydrology Community Discharge (SWOT SHCQ)](https://www.hydroshare.org/resource/465672b67a6d467384eba0724ca979ea/)

**Expected behavior**
I expect the resource to be deleted from the collection

**Additional information**
I did some investigation, there is an error message in the response.
```
{"status": "error", 
"msg": "'NoneType' object has no attribute 'delete'", 
"metadata_status": "Insufficient to make public", 
"new_coverage_list": []}
```
It looks like this happens when there is missing relational metadata, I'd wager either of these delete causes could be the culprit and both should be handled better.  https://github.com/hydroshare/hydroshare/blob/develop/hs_collection_resource/views.py#L141-L147

If the relation metadata that we want to delete is already deleted, then move on.  Also, while in there, please update the log statement to include the stacktrace with `logger.exception` instead of `logger.error` https://github.com/hydroshare/hydroshare/blob/develop/hs_collection_resource/views.py#L207

**Comment by devincowan** [comment]
Interesting, looks like both of the resource have `isPartOf` metadata but neither of them have `hasPart`

```
>>> resources = BaseResource.objects.filter(collections__short_id='d9553a8f9592463984e9a2a631759501').all()
>>> for res in resources:
...     print(res)
...     for rel in res.metadata.relations.all():
...             print(rel)
...
Example Data II For Surface Water and Ocean Topography Hydrology Community Discharge (SWOT SHCQ)
isPartOf Coss, S. (2023). Surface Water and Ocean Topography  Hydrology Community Discharge (SWOT SHCQ), HydroShare, http://www.hydroshare.org/resource/96f1928c95834539ba260ab65ea8db8e
Example Data For Surface Water and Ocean Topography Hydrology Community Discharge (SWOT SHCQ)V2.1
isPartOf Coss, S. (2023). Surface Water and Ocean Topography  Hydrology Community Discharge (SWOT SHCQ), HydroShare, http://www.hydroshare.org/resource/96f1928c95834539ba260ab65ea8db8e
isVersionOf Coss, S. (2023). Example Data For Surface Water and Ocean Topography Hydrology Community Discharge (SWOT SHCQ), HydroShare, http://www.hydroshare.org/resource/0e005f7c51d542999b32aecb69ff9b87
isPartOf Coss, S. (2023). Surface Water and Ocean Topography  Hydrology Community Discharge (SWOT SHCQ), HydroShare, http://www.hydroshare.org/resource/38feeef698ca484b907b7b3eb84ad05b
```

so Scott is probably right that the issue is this line:
https://github.com/hydroshare/hydroshare/blob/develop/hs_collection_resource/views.py#L141
I'll try to replicate locally and fix.

**Comment by devincowan** [comment]
Okay I was able to reproduce this.
Occurs when a new version of a collection is created, the resources in the collection don't have isPartOf relations pointing to the new collection version.
I will update

**Comment by devincowan** [comment]
turns out there are **784** metadata relations (isPartOf or hasPart) missing currently on prod

**Comment by devincowan** [comment]
released in 2.5.3 hotfix today
